### PR TITLE
[Builtin] Make 'makeKnown' receive the term representation of the value

### PR DESCRIPTION
**Pull request by effectfully**

### CODE REVIEW

File: plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs [hunk 1]
```diff
@@ -459,24 +459,27 @@ We use @default@ for providing instances for built-in types instead of @Deriving
 latter breaks on @m a@
 -}
 
+-- TODO: in @makeKnown@ @cause@ and @term@ can be different, while in @readKnown@ they must be the
+-- same. We need to check if making them different in @readKnown@ is feasiable and can simplify
+-- things such as error handling in the CEK machine.
```
Comment (effectfully) [inline plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs:464]: Gonna try this next.

**Comment by effectfully** [comment]
!benchmark

File: plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs (hunk 1 above)
Comment (effectfully) [reply plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs:464]: Seems to have [worked](https://github.com/effectfully/plutus-prototype/commit/baaa34f40882a4aad44d0897a9ea5943631ad367) really well. If we're going to merge what's in this PR, then we should merge what's in that commit too. @michaelpj

**Comment by michaelpj** [comment]
I fear the benchmarking job is still broken :( Although in this case it didn't work because it doesn't work on forks.

File: plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs (hunk 1 above)
Comment (michaelpj) [reply plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs:464]: I agree, it looks great, add it to this PR.

**Comment by michaelpj** [comment]
This looks great.

**Review by michaelpj — APPROVED** [review]
Approving but do include the other commit too.

**Pull request merged**